Sources/Purchasing/Purchases/Purchases.swift

```diff
@@ -827,6 +833,49 @@ public extension Purchases {
 
 }
 
+/**
+ * Clears all cache for the SDK if the user's purchase history is empty.
+ * It can be useful if you do not have your own accounts sytem and rely solely on RevenueCat's anonymous IDs,
+ * and you want to ensure that if a user logs out of their App Store account, they can no longer access their purchases
+ * unless they log back in or restore purchases.
+ */
+private extension Purchases {
+
+    func resetUserIfPurchaseHistoryIsEmptyIfNeeded() {
+        guard self.isAnonymous else { return }
+        guard self.systemInfo.dangerousSettings.disallowSharingAppStoreAccountsForAnonymousIDs,
+        self.systemInfo.dangerousSettings.internalSettings.usesStoreKit2JWS else { return }
+
+        guard #available(iOS 15.0, tvOS 15.0, macOS 13.0, watchOS 9.0, *) else { return }
+
+        _ = Task<Void, Never> {
+
+            // if there are no transactions in StoreKit, but the user has transactions in RevenueCat,
+            // we assume that they've logged out of the account, and we logOut correspondingly.
+            guard let cachedCustomerInfo = customerInfoManager
+                .cachedCustomerInfo(appUserID: identityManager.currentAppUserID) else {
+                return
+            }
+
+            guard !cachedCustomerInfo.allPurchasedProductIdentifiers.isEmpty else {
+                return
+            }
+
+            let hasTransactions = await self.customerInfoManager.transactionFetcher.firstVerifiedTransaction != nil
```

Comment:
Just to make sure but this also returns expired purchases right? Just to make sure we are not logging anonymous users out if their subscriptions expire (shouldn't be a big problem, but we shouldn't create accounts if not needed)

Reply:
yeah, it returns full history including expired auto-renewable subscriptions, non-consumables and consumables with unfinished transactions